aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -171,10 +171,48 @@ def should_fetch_custom_tags():
     return os.environ.get("DD_FETCH_LAMBDA_TAGS", "false").lower() == "true"
 
 
-def sanitize_aws_tag_string(raw_string):
+_other_chars = r"\w:\-\.\/"
+Sanitize = re.compile(r"[^%s]" % _other_chars, re.UNICODE).sub
+Dedupe = re.compile(r"_+", re.UNICODE).sub
+FixInit = re.compile(r"^[_\d]*", re.UNICODE).sub
+FirstCapRe = re.compile("(.)([A-Z][a-z]+)")
+AllCapRe = re.compile("([a-z0-9])([A-Z])")
+
+
+def convert_camel_case_to_underscore(string):
+    """
+    Convert from CamelCase to camel_case
+    """
+    global FirstCapRe, AllCapRe
+    string = FirstCapRe.sub(r"\1_\2", string)
+    return AllCapRe.sub(r"\1_\2", string).lower()
+
+
+def sanitize_aws_tag_string(tag):
     """Convert characters banned from DD but allowed in AWS tags to underscores
     """
-    return re.sub(r"[\+\-=\.:/@]", "_", raw_string)
+    global Sanitize, Dedupe, FixInit
+
+    # 1. Convert camel case to underscores
+    # 2. Replaces colons with _
+    # 3. Convert to all lowercase unicode string
+    # 4. Convert bad characters to underscores
+    # 5. Dedupe contiguous underscores
+    # 6. Remove initial underscores/digits such that the string
+    #    starts with an alpha char
+    #    FIXME: tag normalization incorrectly supports tags starting
+    #    with a ':', but this behavior should be phased out in future
+    #    as it results in unqueryable data.  See dogweb/#11193
+    # 7. Truncate to 200 characters
+    # 8. Strip trailing underscores
+    tag = convert_camel_case_to_underscore(tag)
+    tag = tag.replace(":", "_")
```

Comment:
I think we don't want to replace every colon with an underscore since we don't do that in crawlers. If I'm reading the code [here](https://github.com/DataDog/dogweb/blob/prod/integration/amazon_web_services/utils/common.py#L106-L128) right, the crawlers never replace the colons in tag values but do replace the colons in tag keys if the `aws_tags_replace_colons` FF is enabled, which it is by default for new orgs. Since it's the default for new orgs and we can't check the FF from the forwarder I think it would make sense to make that the behavior here and maybe at some point add an env var that lets customers control the behavior like the FF if customers ask for it.

Reply:
Hmm, good point, I'll try and update the logic.